c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -225,6 +227,18 @@ fn deconstruct_hir_ty<'a, 'tcx>(
                 type_args
             })
         }
+        (&ty::TyKind::FnPtr(sig), &hir::TyKind::BareFn(bfnty)) => {
+            let args = sig.skip_binder().inputs_and_output;
+            if args.len() != bfnty.decl.inputs.len() + 1 {
+                panic!("mismatched number of function inputs for {sig:?} and {bfnty:?}");
+            }
+
+            let mut v: Vec<_> = bfnty.decl.inputs.iter().collect();
+            if let FnRetTy::Return(return_ty) = bfnty.decl.output {
+                v.push(return_ty)
+            }
```

Comment:
Why's it okay for the length of `v` to vary here?  Is the caller doing a `zip` and we're relying on the implicit truncation?

If we were to emit a rewrite on the implicit `()` return type of some function pointer, will the rewrite properly fail, due to not having anywhere to apply the rewrite?